context_chat_backend/controller.py

```diff
@@ -96,7 +75,26 @@ def _():
 	return JSONResponse(content={'status': 'ok'}, status_code=200)
 
 
+@app.post('/init')
+async def _():
+	async def update_progress(progress: int):
+		await ocs_call(
+			method='PUT',
+			path=f'/ocs/v1.php/apps/app_api/apps/status/{getenv("APP_ID")}',
+			json_data={ 'progress': min(100, progress) },
+		)
+
+	if not app.extra.get('ENABLED', False):
+		await download_all_models(app, update_progress)
```

Comment:
please use FastAPI BackgroundTasks in which you will download models in the background

Python's `async` is a blocking call in this context, reply for "/init" will not be received on NC side..

Reply:
Yes, can confirm this issue, finally received time-out error on `/init`

Reply:
that did the trick! How to know if async is blocking in a certain context though?